cdap-ui/app/cdap/components/HttpExecutor/store/HttpExecutorStore.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {combineReducers, createStore} from 'redux';
+import HttpExecutorActions from 'components/HttpExecutor/store/HttpExecutorActions';
+import shortid from 'shortid';
+
+const defaultAction = {
+  action : '',
+  payload : {}
+};
+
+const defaultInitialState = {
+  method: 'GET',
+  path: '',
+  body: '',
+  headers: {
+    pairs: [{
+      key: '',
+      value: '',
+      uniqueId: shortid.generate()
+    }]
+  },
+  response: null,
+  statusCode: 0,
+  loading: false,
+  activeTab: 0
+};
+
+const http = (state = defaultInitialState, action = defaultAction) => {
+  let stateCopy;
+
+  switch (action.type) {
+    case HttpExecutorActions.setMethod:
+      stateCopy = Object.assign({}, state, {
```

Comment:
Now that since we have started using object spread operator. This could be simplified with that.

```
return {
  ...state,
  method: action.payload.method,
  activeTab: ...
};
```

Just a suggestion. Might make it simpler.